usr/lib/cca_stdll/cca_specific.c

```diff
@@ -9261,6 +9274,598 @@ static CK_RV import_ec_pubkey(STDLL_TokData_t *tokdata, TEMPLATE *pub_templ)
+        }
+
+        /* Build key_value_structure */
+        key_value_structure_length = CCA_KEY_VALUE_STRUCT_SIZE;
+        rc = build_ibm_dilithium_import_key_value_struct(
+                                TRUE, oid, priv_templ,
+                                (unsigned char *)&key_value_structure,
+                                &key_value_structure_length);
+        if (rc != CKR_OK) {
+            TRACE_ERROR("build_ibm_dilithium_import_key_value_struct failed: "
+                        "0x%lx\n", rc);
+            return rc;
+        }
+
+        /* Build key token */
+        rule_array_count = 2;
+        memcpy(rule_array, "QSA-PAIRU-DIGSIG", (size_t)(CCA_KEYWORD_SIZE * 2));
+        private_key_name_length = 0;
+        key_token_length = CCA_KEY_TOKEN_SIZE;
+
+        USE_CCA_ADAPTER_START(tokdata, return_code, reason_code)
+            dll_CSNDPKB(&return_code, &reason_code,
+                        &exit_data_len, exit_data,
+                        &rule_array_count, rule_array,
+                        &key_value_structure_length, key_value_structure,
+                        &private_key_name_length, private_key_name,
+                        &param1, param2, &param1, param2, &param1, param2,
+                        &param1, param2, &param1, param2,
+                        &key_token_length,
+                        key_token);
+        USE_CCA_ADAPTER_END(tokdata, return_code, reason_code)
+
+        if (return_code != CCA_SUCCESS) {
+            TRACE_ERROR("CSNDPKB (QSA KEY TOKEN BUILD) failed. return:%ld,"
+                        " reason:%ld\n", return_code, reason_code);
+            return CKR_FUNCTION_FAILED;
+        }
+
+        /* Now import the PKA key token */
+        rule_array_count = 1;
+        memcpy(rule_array, "QSA     ", (size_t)(CCA_KEYWORD_SIZE));
+        key_token_length = CCA_KEY_TOKEN_SIZE;
+        target_key_token_length = CCA_KEY_TOKEN_SIZE;
+
+        USE_CCA_ADAPTER_START(tokdata, return_code, reason_code)
+            dll_CSNDPKI(&return_code, &reason_code, NULL, NULL,
+                        &rule_array_count, rule_array,
+                        &key_token_length, key_token,
+                        transport_key_identifier,
+                        &target_key_token_length, target_key_token);
+        USE_CCA_ADAPTER_END(tokdata, return_code, reason_code)
+
+        if (return_code != CCA_SUCCESS) {
+            TRACE_ERROR("CSNDPKI (QSA KEY TOKEN IMPORT) failed. "
+                        "return:%ld, reason:%ld\n",
+                        return_code, reason_code);
+            return CKR_FUNCTION_FAILED;
+        }
+
+        if (analyse_cca_key_token(target_key_token, target_key_token_length,
+                                  &token_type, &token_keybitsize,
+                                  &mkvp) == FALSE ||
+            mkvp == NULL) {
+            TRACE_ERROR("Invalid/unknown cca token has been imported\n");
+            return CKR_FUNCTION_FAILED;
+        }
```

Comment:
Is this possible? Successfully importing a key, but afterwards we realize that it's invalid/unknown?

Reply:
It might be a key token that `analyse_cca_key_token()` does not support. 
In any case, we need to handle the error case, even if it is unlikely that it happens.